harness/determined/common/api/authentication.py

```diff
@@ -62,139 +69,79 @@ def default_load_user_password(
-        if token is not None and not _is_token_valid(self.master_address, token, cert):
-            self.token_store.drop_user(session_user)
-            token = None
-
-        # Special case: use token provided from the container environment if:
-        # - No token was obtained from the token store already,
-        # - There is a token available from the container environment, and
-        # - No user was explicitly requested, or the requested user matches the token available
-        #   in the container environment.
-        if (
-            token is None
-            and util.get_det_username_from_env() is not None
-            and util.get_det_user_token_from_env() is not None
-            and requested_user in (None, util.get_det_username_from_env())
-        ):
-            session_user = util.get_det_username_from_env()
-            assert session_user
-            token = util.get_det_user_token_from_env()
-
-        if token is not None:
-            return UsernameTokenPair(session_user, token)
-
-        # We'll need to create a new token, so we'll need a password. If there was no requested
-        # user and we ended up falling back to the default username `determined`, then we can fall
-        # back to the default login as well. Otherwise, ask the user for their password.
-        fallback_to_default = password is None and session_user == constants.DEFAULT_DETERMINED_USER
-        if fallback_to_default:
-            password = constants.DEFAULT_DETERMINED_PASSWORD
-
-        if password is None:
-            password = getpass.getpass("Password for user '{}': ".format(session_user))
-
-        try:
-            token = do_login(self.master_address, session_user, password, cert)
-        except api.errors.ForbiddenException:
-            if fallback_to_default:
-                raise api.errors.UnauthenticatedException(username=session_user)
-            raise
-
-        self.token_store.set_token(session_user, token)
-
-        return UsernameTokenPair(session_user, token)
-
-    def is_user_active(self, username: str) -> bool:
-        return self.token_store.get_active_user() == username
-
-    def get_session_user(self) -> str:
-        """
-        Returns the session user for the current session. If there is no active
-        session, then an UnauthenticatedException will be raised.
-        """
-        return self.session.username
-
-    def get_session_token(self, must: bool = True) -> str:
-        """
-        Returns the authentication token for the session user. If there is no
-        active session, then an UnauthenticatedException will be raised.
-        """
-        if self.session is None:
-            if must:
-                raise api.errors.UnauthenticatedException(username="")
-            else:
-                return ""
-        return self.session.token
-
-
-def do_login(
-    master_address: str,
-    username: str,
-    password: str,
+def login_with_cache(
```

Comment:
When should this set the `active_user` in the token store? Currently, don't think it ever does.

Reply:
I think of that as being an action explicitly tied to `det user login`, since IIRC that's the only way we currently have to "store" which user you are logged in as.

`login_with_cache()` is also used when somebody sets a temporary user via `det -u admin user create` or something like that.  So `login_with_cache()` just doesn't have the context to know if it should set the active user.

Reply:
> the only way we currently have to "store" which user you are logged in as.

Per our earlier definition of "log in", I may be logged in as many users. What is `active_user` supposed to represent, and who should have the responsibility of setting it? Is the CLI supposed to know about the token store, or is that an implementation detail auth should abstract away?

Reply:
At some point (not in this PR) we've decided the CLI _is_ supposed to know about the TokenStore. Which is weird to me, but certainly outside the scope of this PR.

Good to resolve.